src/RandMsg.chpl

```diff
@@ -438,6 +438,100 @@ module RandMsg
         }
     }
 
+    inline proc gammaGenerator(kArg: real, ref rs) throws {
+        // TODO fill this in i guess
+        if kArg == 1.0 {
+            // Okay for now im just gonna pick one but we should use method here somehow
+            // maybe not bc zig requires two generators, so inverse is simpler
+            return standardExponentialInvCDF(1, rs)[0];
+        }
+        else if kArg == 0.0 {
+            return 0.0;
+        }
+        else if kArg < 1.0 {  // TODO ask the chpl team about the linter being weird
+            var count = 0;
+            while count < 10000000 {
+                var U = rs.next(0, 1);
+                var V = standardExponentialInvCDF(1, rs)[0];
+                if U <= (1.0 - kArg) {
+                    //can these be more effiection 
+                    var X = U ** (1.0 / kArg);
+                    if X <= V {
+                        writeln("XXX   count is ", count);
+                        return X;
+                    }
+                }
+                else {
+                    var Y = -log((1.0 - U) / kArg);
+                    //can these be more effiection 
+                    var X = (1.0 - kArg + kArg * Y) ** (1.0 / kArg);
+                    if X <= (V + Y) {
+                        writeln("XXX   count is ", count);
+                        return X;
+                    }
+                }
+                count+= 1;
+            }
+            writeln("\n\n\n\n\n\n\n\n\n\n\n\n\n\n");
+            writeln("OMG FIRST LOOP IS BROKE!!");
+
+            return -1.0;  // we failed 10000 times in a row which should practically never happen
+
+        } 
+        else {
+            var b = kArg - 1/3;
+            var c = 1/sqrt(9 * b);
+            var X = 0.0;
+            var V = -1.0;
```

Comment:
I think this is the bug.  `var V = -1.0;` needs to go inside the while loop.  Otherwise, it does `V * V * V` over and over again which eventually goes to infinity.

Reply:
On a first analysis, it looks like the V = -1.0 line was inserted so that the "while V <= 0" loop almost immediately below will execute at least once.  But that could be done via a do-while loop, as is done in the python.